test/Dialect/Moore/types.mlir

```diff
@@ -46,34 +32,34 @@ func.func @RealTypes(
 
 // CHECK-LABEL: func @DimTypes(
 func.func @DimTypes(
-  // CHECK-SAME: %arg0: !moore.packed<unsized<bit>>,
-  // CHECK-SAME: %arg1: !moore.packed<range<bit, 4:-5>>,
-  %arg0: !moore.packed<unsized<bit>>,
-  %arg1: !moore.packed<range<bit, 4:-5>>,
-  // CHECK-SAME: %arg2: !moore.unpacked<unsized<bit>>,
-  // CHECK-SAME: %arg3: !moore.unpacked<array<bit, 42>>,
-  // CHECK-SAME: %arg4: !moore.unpacked<range<bit, 4:-5>>,
-  // CHECK-SAME: %arg5: !moore.unpacked<assoc<bit>>,
-  // CHECK-SAME: %arg6: !moore.unpacked<assoc<bit, string>>,
-  // CHECK-SAME: %arg7: !moore.unpacked<queue<bit>>,
-  // CHECK-SAME: %arg8: !moore.unpacked<queue<bit, 9001>>
-  %arg2: !moore.unpacked<unsized<bit>>,
-  %arg3: !moore.unpacked<array<bit, 42>>,
-  %arg4: !moore.unpacked<range<bit, 4:-5>>,
-  %arg5: !moore.unpacked<assoc<bit>>,
-  %arg6: !moore.unpacked<assoc<bit, string>>,
-  %arg7: !moore.unpacked<queue<bit>>,
-  %arg8: !moore.unpacked<queue<bit, 9001>>
+  // CHECK-SAME: %arg0: !moore.packed<unsized<i1>>,
+  // CHECK-SAME: %arg1: !moore.packed<range<i1, 4:-5>>,
+  %arg0: !moore.packed<unsized<i1>>,
+  %arg1: !moore.packed<range<i1, 4:-5>>,
+  // CHECK-SAME: %arg2: !moore.unpacked<unsized<i1>>,
+  // CHECK-SAME: %arg3: !moore.unpacked<array<i1, 42>>,
+  // CHECK-SAME: %arg4: !moore.unpacked<range<i1, 4:-5>>,
+  // CHECK-SAME: %arg5: !moore.unpacked<assoc<i1>>,
+  // CHECK-SAME: %arg6: !moore.unpacked<assoc<i1, string>>,
+  // CHECK-SAME: %arg7: !moore.unpacked<queue<i1>>,
+  // CHECK-SAME: %arg8: !moore.unpacked<queue<i1, 9001>>
+  %arg2: !moore.unpacked<unsized<i1>>,
+  %arg3: !moore.unpacked<array<i1, 42>>,
+  %arg4: !moore.unpacked<range<i1, 4:-5>>,
+  %arg5: !moore.unpacked<assoc<i1>>,
+  %arg6: !moore.unpacked<assoc<i1, string>>,
+  %arg7: !moore.unpacked<queue<i1>>,
+  %arg8: !moore.unpacked<queue<i1, 9001>>
```

Comment:
Nit: the mnemonic `i1` could be a bit confusing since it could be the built-in integer type or the Moore type here if you don't know what the parser does.

Reply:
Yeah that is very true. I was thinking of using `b1` and `l1` to distinguish this from the builtin integers. On the other hand, we might want to eventually replace `IntType` with a `LogicType` for `l42` and use the builtin integers for `i42`, so the `i1` might be easier to update in that case.

Reply:
Any preferences @maerhart :grin: ?

Reply:
If the plan is to replace the two-valued Moore IntType with the built-in, I think keeping it as `i1` for now makes a lot of sense. 🙂